File: pypp_cli/test_dir/python/src/loops/for_examples.py
def pseudo_fn(integers: list[int], strings: list[str]):
    # typical
    for integer in integers:
        print(integer)
    # with range()
    for i in range(len(integers)):
        print(i)
    for i in range(len(integers), 5, -2):
        print(i)
    # with enumerate()
    for i, val in enumerate(integers):
        print(i, val)
    # with zip()
    for integer, string in zip(integers, strings):
        print(integer, string)
    # with reversed()
    for integer in reversed(integers):
        print(integer)

File: pypp_cli/test_dir/cpp/src/loops/for_examples.cpp
#include "loops/for_examples.h"
#include "py_enumerate.h"
#include "py_reversed.h"
#include "py_zip.h"
#include "pypp_util/print.h"

namespace me {
void pseudo_fn(pypp::PyList<int> &integers,
               pypp::PyList<pypp::PyStr> &strings) {
    for (const auto &integer : integers) {
        pypp::print(integer);
    }
    for (int i = 0; i < integers.len(); i += 1) {
        pypp::print(i);
    }
    for (int i = integers.len(); i < 5; i += -2) {
        pypp::print(i);
    }
    for (const auto &[i, val] : pypp::PyEnumerate(integers)) {
        pypp::print(i, val);
    }
    for (const auto &[integer, string] : pypp::PyZip(integers, strings)) {
        pypp::print(integer, string);
    }
    for (const auto &integer : pypp::PyReversed(integers)) {
        pypp::print(integer);
    }
    for (const auto &c : pypp::PyStr("abcdefg")) {
        pypp::print(c);
    }
}

} // namespace me